### PR TITLE
Ug update

### LINKED ISSUE #164: [PE-D][Tester C] jar file in User Guide links to AB3 instead of ConTActs
The .jar file link in the Installation Guide of the User Guide links to the se-edu-addressbook-level3 github release instead of your team's release. 

Perhaps you could update the link, to link to your team's product instead.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: rayray39/ped#1

### LINKED ISSUE #167: [PE-D][Tester A] Add person example in UG under installation guide not correct
add command shown is incorrect

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: Justincjr/ped#2

### LINKED ISSUE #168: [PE-D][Tester C] User Guide references addressbook which may be confusing
The user guide references the word 'addressbook' in multiple areas. As a user, it may be confusing since your product is called ConTActs and the term addressbook may refer to something unknown.

Perhaps you could consider one of the following improvements,
1. replace all 'addressbook' with 'ConTActs', as that is the name of your product.
2. have a one-liner to explain that addressbook just means the contact list of ConTActs.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: rayray39/ped#2

### LINKED ISSUE #169: [PE-D][Tester A] AddressBook is used instead of app name ConTActs in UG
ConTActs was used initially
Then subsequently AddressBook is used instead of ConTActs which can be very confusing for users

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: Justincjr/ped#3

### LINKED ISSUE #178: [PE-D][Tester E] Sample command given in the UserGuide fails
## Description
The sample `add` command from **UserGuide**, page **2** fails.

## Steps to reproduce
1. Copy the sample `add` command from UserGuide: `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01`.
2. Paste the command into the app.

You should be able to observe that the sample command fails.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: wuzengfu/ped#2

### LINKED ISSUE #187: [PE-D][Tester D] `mark` command resets my `find` filter
Reproduction:
1. `find david`
2. `mark 1 tut/2`
3. Notice how you are thrown back to the full listing

This behavior is quite disruptive when I have a lot of students and every time I find a specific student which I want to mark, it throws me back to the full listing, making me have to re-find my student to review the changes or do further actions.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: shotnothing/ped#9

### LINKED ISSUE #190: [PE-D][Tester C] Edited name allows numbers
when editing a person, the name of the person is allowed to be purely in integers. Considering that people rarely have names in purely integer formats, perhaps you might consider making names to take in only letters. 

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: rayray39/ped#6

### LINKED ISSUE #191: [PE-D][Tester D] Help window's `Copy URL` button is not clear as to what I am copying
Reproduction:
1. `help`  

Button is so far away that most reasonable people will not make the connection.
I suggest either putting the button below or (better option) making the link label a hyperlink.

-------------
Labels: `type.FeatureFlaw` `severity.VeryLow`
original: shotnothing/ped#10

### LINKED ISSUE #196: [PE-D][Tester D] Able to click and focus onto cards in the UI, but nothing occurs, which is misleading
In the application's user interface, certain cards are clickable, allowing users to focus on them by clicking, but no action or response is triggered. This behavior can be misleading, as users may expect an action or response when interacting with these clickable cards.

-------------
Labels: `type.FeatureFlaw` `severity.VeryLow`
original: shotnothing/ped#14

### LINKED ISSUE #200: [PE-D][Tester D] Sorting does not take into account unmarked tutorial sessions
Reproduction:
1. Start from clean data JSON
2. `mark 3 tut/4` (i.e. Charlotte attended tut 4)
3. `sort 1 tut/4`
4. `sort -1 tut/4`
5. Notice that Charlotte entry doesn't move, stays at 1. Nothing actually moves.

I think that the UG description of `sort` does not specify what ascending or descending means in the context of a 3-state attendance (present, absent and not-taken-place). 

Thus, as a user, I would assume sorting in descending order would give me results the inverse of ascending. 

I think this could be considered a documentation, functionality and feature bug, or some combination of the three, depending on the angle taken

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: shotnothing/ped#18

### LINKED ISSUE #205: [PE-D][Tester E] Incosistent UserGuide documentation on sort command
## Description
The `sort` command format specified in the UserGuide is `sort ORDER [n/][i/][tut/TUTORIAL]`. Note that there is **no space** in between the fields, this may confuse the user as they may try to enter command such as `sort 1 n/i/` instead of `sort 1 n/ i/`. However, the other commands such as `find` indeed have spaces in between the fields. Indeed, if a user follows the command format as specfied in the UserGuide, it fails.

## Steps to reproduce
1. Enter the command `sort 1 n/i/`.

You should be able to observe that the command fails which contradicate to the UserGuide given by the team.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: wuzengfu/ped#7

### LINKED ISSUE #207: [PE-D][Tester C] Unclear of which fields can be duplicated
When trying to add john into the current list of contacts, the error indicates that john already exists in the contacts. Upon closer inspection, it seems that all the fields (name, number, id, email) must be unique for every person. 

Perhaps you could indicate in the user guide which fields must be unique and which fields can be duplicated across the contacts. Either that or you could make the error message more descriptive to tell the user which field must be changed because it already exists. If the user makes an error in typing the phone number or ID for eg. and that number already exists in the contacts, the error message is not descriptive enough to let the user know he needs to change the phone number/id. 

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: rayray39/ped#7

### LINKED ISSUE #208: [PE-D][Tester E] Inconsistent optional fields in UserGuide for sort command
## Description
As given by the UserGuide that `[FIELD]` means this field is **optional**. However, the `sort` command in the UserGuide specifies all fields as **optional**. When a user tries to follow the format given by team without including any field, the command fails.

## Steps to reproduce
1. Enter the command `sort 1`.

You should be able to observe an error message given **although the command format matches the UserGuide provided**.

## Suggestion
The UserGuide documentation can state that a user must provide at least one field.

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: wuzengfu/ped#8

### LINKED ISSUE #218: [PE-D][Tester B] Not enough screenshots in UG
No screenshots were provided for how the mark and unmark features will look like when the tutorials are marked.

-------------
Labels: `severity.Medium` `type.DocumentationBug`
original: dez2003/ped#9

### LINKED ISSUE #221: [PE-D][Tester A] Sort feature not explaiend well in UG
The sort feature in the UG does not explain how the names, ID and tutorials are sorted. As a user, I have no clue what it is sorting and I have no idea how to take advantage of this feature

-------------
Labels: `severity.High` `type.DocumentationBug`
original: Justincjr/ped#11

### LINKED ISSUE #226: [PE-D][Tester B] Invalid command example in UG
`add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123,#01-01` is an invalid command example in the UG since `a/` isn't an accepted prefix in their product.

-------------
Labels: `severity.High` `type.DocumentationBug`
original: dez2003/ped#11

### LINKED ISSUE #229: [PE-D][Tester B] Clearer Explanation about the Wildcard
It does mention what the wildcard `*` is deeper in the UG, but since the first mention of the wildcard is in the mark feature and it doesn't explain what it means, I think that the image below should be mentioned before introducing the features that can use wildcards.

This is where it first mentions wildcards:

-------------
Labels: `severity.Low` `type.DocumentationBug`
original: dez2003/ped#14

### PR DESCRIPTION
**Pull request by raageth**
- close #229 
- close #226 
- close #218 
- close #208 
- close #205 
- close #178 
- close #169
- close #167
- close #168 
- close #164 
--- 
- close #191 
- close #190 
- close #196 
- close #221 
- close #200 
- close #207 
- close #187

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -75,168 +75,203 @@ Shows a message explaning how to access the help page.
...
-### Resetting person's attendance : `reset`
+Visual Effect:
 
-Resets the attendance of the person by the index number.
+- After the command `mark 1 tut/1`, tutorial box 1 of the 1st student will turn <span style="color:green">green</span>.
 
-Format: `reset INDEX tut/TUTORIAL`
+  |Before|After|
+  |---|---|
+  |![Before](images/MarkBefore.png)|![After](images/MarkAfter.png)|
```
Comment (raageth) [inline docs/UserGuide.md:171]: fixes #218

File: docs/UserGuide.md [hunk 2]
```diff
@@ -75,168 +75,203 @@ Shows a message explaning how to access the help page.
...
+- `mark * tut/1` marks all students as attended for tutorial 1.
+- `unmark * tut/1` marks all students as absent for tutorial 1.
+- `reset * tut/1` resets the attendance of all students for tutorial 1.
+</div>
+
+### Sorting students : `sort`
+
+Sorts the displayed list of students by either name, student id or tutorial attendance. The sorting order will be maintained while edits to the list are made.
+
+Format: `sort ORDER [n/] [i/] [tut/TUTORIAL]`
```
Comment (raageth) [inline docs/UserGuide.md:236]: fix #205 
- add spaces between flags

File: docs/UserGuide.md [hunk 3]
```diff
@@ -75,168 +75,203 @@ Shows a message explaning how to access the help page.
...
 
 - `ORDER` indicates whether the sorted list is **ascending** or **descending**.
 - **Ascending** is represented with integer 1.
 - **Descending** is represented with integer -1.
 - `n/` indicates sorting according to name.
 - `i/` indicates sorting according to student id.
 - `tut/TUTORIAL` indicates sorting according to tutorial attendance for a specific tutorial.
+- At least one sorting criterion `[n/]`, `[i/]`, `[tut/TUTORIAL]` must be present. `sort ORDER` is an invalid command.
```
Comment (raageth) [inline docs/UserGuide.md:244]: fix #208 
- state that at least one field must be present

File: docs/UserGuide.md [hunk 4]
```diff
@@ -18,24 +18,24 @@ If you can type fast, ConTActs can get your contact management tasks done faster
...
    ![Ui](images/Ui.png)
 
 1. Type the command in the command box and press Enter to execute it. e.g. typing **`help`** and pressing Enter will open the help window.<br>
    Some example commands you can try:
 
-   - `list` : Lists all contacts.
+   - `list` : Lists all students in the contact list.
 
-   - `add n/John Doe p/98765432 e/johnd@example.com a/John street, block 123, #01-01` : Adds a contact named `John Doe` to the Address Book.
+   - `add n/John Doe i/E0000000 p/98765432 e/johnd@example.com` : Adds a student named `John Doe` to the contact list.
```
Comment (raageth) [inline docs/UserGuide.md:34]: fix #226 fix #167 fix #178

- remove address field

File: docs/UserGuide.md [hunk 5]
```diff
@@ -75,168 +75,203 @@ Shows a message explaning how to access the help page.
...
-Sorts the displayed list of persons by either name, student id or tutorial attendance. The sorting order will be maintained while edits to the list are made.
+<div markdown="block" class="alert alert-info">
 
-Format: `sort ORDER [n/][i/][tut/TUTORIAL]`
+**:information_source: Faster attendance updating:**<br>
+You can use the `mark`, `unmark` and `reset` commands with the wildcard `*` to update the attendance of all currently displayed students in the contact list at once.
+- `mark * tut/1` marks all students as attended for tutorial 1.
+- `unmark * tut/1` marks all students as absent for tutorial 1.
+- `reset * tut/1` resets the attendance of all students for tutorial 1.
+</div>
```
Comment (raageth) [inline docs/UserGuide.md:230]: fix #229 

- add block to explain wildcard

**Review by sayglenn — APPROVED** [review]
LGTM!

**Comment by sayglenn** [comment]
You can change these as well.

#180
#179
#175
#173
#171

File: docs/UserGuide.md
Comment (Domokunx) [inline docs/UserGuide.md:1]: Very last line, "Back to top" link does not redirect to top, but to command summary instead.

Comment (Domokunx) [inline docs/UserGuide.md:1]: Line 72 typo "explaning"

**Review by Domokunx — COMMENTED** [review]
cosmetic

File: docs/UserGuide.md (hunk 3 above)
Comment (yockcheng) [reply docs/UserGuide.md:253]: Should be "only one". We don't support multiple criterion.

File: docs/UserGuide.md [hunk 6]
```diff
@@ -69,174 +69,219 @@ If you can type fast, ConTActs can get your contact management tasks done faster
...
 
-- Resets the attendance of the person at the specified `INDEX`. The index refers to the index number shown in the displayed person list. The index **must be a positive integer** 1, 2, 3, …​ or a **wildcard** *.
+- Resets the attendance of the student at the specified `INDEX`. The index refers to the index number shown in the displayed student list. The index **must be a positive integer** (1, 2, 3, ...).
 - `TUTORIAL` can be in the format of:
-    - A positive number between 1-12 `eg. 1`.
-    - A list of numbers `eg. [1,3,5]`.
-    - A range of two numbers `eg. 3-6`.
+    - A positive number between 1 - 12 (inclusive) e.g. `1`.
+    - A list of numbers e.g. `[1,3,5]`.
+    - A range of two numbers e.g. `3-6`.
```
Comment (Domokunx) [inline docs/UserGuide.md:210]: Should we specify positive numbers in ascending order or is that too much?

**Review by sayglenn — APPROVED** [review]
LGTM!

**Pull request merged**